synapse/server.py

```diff
@@ -286,6 +286,10 @@ def setup(self) -> None:
         if self.config.run_background_tasks:
             self.setup_background_tasks()
 
+    def start_listening(self) -> None:
+        """Start the HTTP, manhole, metrics, etc listeners"""
+        pass
```

Comment:
I find it a bit odd that the default implementation is a no-op, given the doc comment. What do you think about making this an abstract method and keeping the no-op version in admin cmd?

Reply:
personally, it seems fairly reasonable that the default be a no-op - I basically see it as a hook that the derived classes can override. I could make it clearer in the docstring?

don't feel that strongly though, so let me know what you think.

Reply:
I guess it just feels slightly odd that `start_listening()` doesn't do the right thing by default (which it feels like it could if we deduplicate the code between master and worker apps), the fact that the admin cmd server is no-oping it is a special thing that its doing that I'd expect to be specifically called out in its definition.

It doesn't really matter though, given we're not really adding new server objects.